docs/reference/eql/eql.asciidoc

```diff
@@ -792,6 +792,44 @@ results by using <<get-async-eql-status-api,get async EQL status API>>.
 You can also manually delete saved synchronous searches using the
 <<delete-async-eql-search-api,delete async EQL search API>>.
 
+[discrete]
+[[eql-circuit-breaker]]
+=== EQL circuit breaker settings
+
+When a <<eql-sequences, sequence>> query is executed, the node handling the query
+needs to keep some structures in memory, which are needed by the algorithm
+implementing the sequence matching. When large amounts of data need to be processed,
+and/or a large amount of matched sequences is requested by the user (by setting the
+<<eql-search-api-params-size, size>> query param), the memory occupied by those
+structures could potentially exceed the available memory of the JVM. This would cause
+an `OutOfMemory` exception which would bring down the node.
+
+To prevent this from happening, a special <<circuit-breaker, circuit breaker>> is used,
+which limits the memory allocation during the execution of a <<eql-sequences, sequence>>
+query. When the breaker is triggered, an `org.elasticsearch.common.breaker.CircuitBreakingException`
+is thrown and a descriptive error message is returned to the user.
```

Comment:
This copy is good, but I wonder if we're over-explaining a bit. I'd consider shortening it:

```suggestion
The EQL circuit breaker limits the amount of memory that sequence queries can
use. The breaker prevents an OutOfMemoryError for queries with a large
<<eql-search-api-params-size,`size`>> or data set.
```

Reply:
@costin What do you think?

Reply:
I'd leave it as is. It didn't strike me as too verbose or too detailed. If the overall style of EQL documentation or that of the circuit breakers in Elasticsearch is more concise/trimmed down, then go with @jrodewig suggestion. Otherwise, my personal opinion is to leave it as is. Circuit breakers are a tricky subject in general imo since the memory calculation algorithm is rather  non-obvious to the regular user. Enough information passed on in the docs helps sometimes clearing out the confusion around them.

Reply:
What @astefan wrote.

Reply:
Thx!